draft-irtf-cfrg-frost.md

```diff
@@ -224,12 +224,49 @@ for more details about each.
 Beyond the core dependencies, the protocol in this document depends on the
 following helper operations:
 
+- Nonce generation, {{dep-nonces}}
 - Schnorr signatures, {{dep-schnorr}};
 - Polynomial operations, {{dep-polynomial}};
 - Encoding operations, {{dep-encoding}};
 - Signature binding {{dep-binding-factor}}, group commitment {{dep-group-commit}}, and challenge computation {{dep-sig-challenge}}
 
-This sections describes these operations in more detail.
+These sections describes these operations in more detail.
+
+## Nonce generation {#dep-nonces}
+
+Straightforward nonce generation is simple:
+
+~~~
+  nonce_generate():
+
+  Outputs: Scalar nonce
+
+  def nonce_generate():
+    return G.RandomNonzeroScalar()
+~~~
+
+This fully relies on the good randomness of the random number generation powering `G.RandomNonZeroScalar()`.
+You may choose to generate randomness to combine with the secret key to hedge against a bad RNG, to create a domain-separated hash function from the ciphersuite hash function `H`:
+
+~~~
+  nonce_generate(secret_share):
+  
+  Inputs:
+  - secret_share, a Scalar in GF(p).
```

Comment:
```suggestion
  - secret_share, a Scalar
```

Also, why is this called `secret_share`, and why does it need to be a Scalar? Can't it just be random bytes?

Reply:
This is to perform `H(randomness, sk_i)` so I agree with @dconnolly's naming

Reply:
OK cool